libraries/lib-time-and-pitch/StaffPad/TimeAndPitch.cpp

```diff
@@ -187,7 +187,7 @@ namespace {
 // wrap a phase value into -PI..PI
 inline float _unwrapPhase(float arg)
 {
-  return arg - std::rint(arg * 0.15915494309f) * 6.283185307f;
+  return arg - int(arg * 0.15915494309f + 0.5f) * 6.283185307f;
```

Comment:
`assert(std::rint(x) == int(x + 0.5f)` fails for values such as `x = -0.7`, i.e., expected is -1, actual is 0.
I think the following would work:
```
  const auto x = arg * 0.15915494309f;
  const auto rounder = x < 0.f && int(-x * 2) % 2 == 1 ? -0.5f : 0.5f;
  return arg - int(arg * 0.15915494309f + rounder) * 6.283185307f;
```
I wonder, though, how this mistake didn't lead to greater errors in your output comparisons 🤔 Maybe now you'd have bit-exactness ?

Reply:
To convince myself I tried the following:
```
inline float _unwrapPhase(float arg)
{
  const auto x = arg * 0.15915494309f;
  const auto rounder = x < 0.f && int(-x * 2) % 2 == 1 ? -0.5f : 0.5f;
  return arg - int(arg * 0.15915494309f + rounder) * 6.283185307f;
}

inline float _unwrapPhaseNaive(float arg)
{
  const auto x = arg * 0.15915494309f;
  const auto rounder = 0.5f;
  return arg - int(arg * 0.15915494309f + rounder) * 6.283185307f;
}
```
and then
```
  assert(_unwrapPhase(twoPi * -0.2) == twoPi * -0.2f);
  assert(_unwrapPhase(twoPi * 0.2) == twoPi * 0.2f);
  assert(_unwrapPhase(twoPi * 0.7) == twoPi * 0.7f - twoPi);
  assert(_unwrapPhase(twoPi * -0.7) == twoPi - twoPi * 0.7f);

  assert(_unwrapPhaseNaive(twoPi * -0.2) == twoPi * -0.2f);
  assert(_unwrapPhaseNaive(twoPi * 0.2) == twoPi * 0.2f);
  assert(_unwrapPhaseNaive(twoPi * 0.7) == twoPi * 0.7f - twoPi);
  assert(_unwrapPhaseNaive(twoPi * -0.7) == twoPi - twoPi * 0.7f);
```
The last assert fails.

Reply:
> Maybe now you'd have bit-exactness ?

No

Reply:
You raise a valid point, though. I will make sure that _unwrapPhase satisfies the post-condition. This is not the same as the exact value though, `rint` results depend on the global state of the application.

Reply:
I've pushed a commit that makes `_unwrapPhase` and `_unwrapPhaseVec` to use exactly the same set of instructions

Reply:
> `rint` results depend on the global state of the application

Would that explain why your comparison tests only showed a -90dB difference, i.e., on your system it was doing the rounding properly ?

Reply:
I'm not sure what you mean.